### PR TITLE
effects: add Mono and Swap effects

### PR DESCRIPTION
**Pull request by jwangsadinata**
This commit adds two different effects, namely `Mono`, that combines the
L/R channels and mix it down into a single channel, and `Swap` that
simply swap the L/R channel outputs.

### CODE REVIEW

**Comment by jwangsadinata** [comment]
@faiface I was having quite a lot of fun with beep and decided to add some effects to your package. These are the basic ones that can be useful, but I might consider helping out in implementing some other non-trivial effects like distortion or bit-crushing to this package when I have some other spare time. Will probably do it in a separate PR tho.

File: effects/mono.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// Mono converts the wrapped Streamer to a mono buffer
+// by downmixing the left and right channels together.
+type Mono struct {
+	Streamer beep.Streamer
+}
+
+// Stream streams the wrapped Streamer while converting the channels into mono.
+func (m *Mono) Stream(samples [][2]float64) (n int, ok bool) {
+	n, ok = m.Streamer.Stream(samples)
+
```
Comment (faiface) [inline effects/mono.go]: unnecessary empty line

File: effects/mono.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	for i := range samples[:n] {
+		mix := samples[i][0] + samples[i][1]
```
Comment (faiface) [inline effects/mono.go]: shouldn't we divide this by 2, so that the total amplitude remains the same?

**Comment by faiface** [comment]
Oh, and btw, have you tried speeding up and slowing down your songs with [ResampleRatio](https://godoc.org/github.com/faiface/beep#ResampleRatio)? That's a lot of fun, because the resampling has a very high quality and it just all sounds so cool when slowed down (or sped up) :D

**Comment by faiface** [comment]
Thanks, this is good. Merge right about now.

**Pull request merged**